### PR TITLE
For #11991 - Show tab count during onboarding

### PR DESCRIPTION
**Pull request by ekager**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 1]
```diff
@@ -225,10 +225,10 @@ class HomeFragment : Fragment() {
         activity.themeManager.applyStatusBarTheme(activity)
 
         view.consumeFrom(requireComponents.core.store, viewLifecycleOwner) {
-            val tabCount = if (currentMode.getCurrentMode() == Mode.Normal) {
-                it.normalTabs.size
-            } else {
+            val tabCount = if (browsingModeManager.mode.isPrivate) {
```
Comment (grigoryk) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:228]: An exhaustive `when` would have prevented this bug, but i guess we can't do that w/ sealed classes?

Comment (grigoryk) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:228]: Actually, you can switch to an exhaustive `when` here, so let's do that instead.

Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:228]: We were just using the wrong way to get the mode. `getCurrentMode()` returns `Mode` -> private, normal, onboarding https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/home/Mode.kt#L21-L24

Now this is just a boolean (which is how we use it everywhere else that doesn't care about onboarding status): https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/browser/browsingmode/BrowsingModeManager.kt#L18

Comment (grigoryk) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:228]: Ah, cool -  i've misread that `if` statemen. That's what i get for writing code, reviewing code and listening to a meeting all at once.

**Pull request merged**